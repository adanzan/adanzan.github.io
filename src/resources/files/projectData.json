[
    {
        "name": "Workout Planner",
        "description": "A workout planner with a GUI.",
        "technologies": "C++, QT",
        "link": "https://github.com/adanzan/workout-planner"
    },
    {
        "name": "Proficiency Tracker",
        "description": "A proficiency tracker for use in a classroom.",
        "technologies": "React.js, Next.js, Firebase",
        "link": "https://github.com/adanzan/proficiency-tracker"
    }
]