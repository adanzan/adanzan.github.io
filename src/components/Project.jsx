/* eslint no-unused-vars: 0 */
import * as React from "react";
import ProjectCard from "./ProjectCard";

import styles from "../styles/Project.module.css";
import projectData from "../resources/files/projectData.json";

export default function Project() {
	const projectCards = [];
	projectData.map((project) => {
		projectCards.push(
			<ProjectCard 
				key={project.name}
				name={project.name} 
				description={project.description}
				image={project.image}
				technologies={project.technologies}
				link={project.link}
			/>);
	});

	return (
		<div id="projects">
			<h2>PROJECTS</h2>
			<div className={styles.projectCards}>
				{projectCards}
			</div>
		</div>);
}