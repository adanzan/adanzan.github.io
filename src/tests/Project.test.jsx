/* eslint no-unused-vars: 0 */
import * as React from "react";
import { render } from "@testing-library/react";

import Project from "../components/Project"; 
import projectData from "../resources/files/projectData.json";

let testRender;


describe("Project component", () => {
	beforeEach(() => {
		testRender = render(<Project />);
	});
	test("matches snapshot", () => {
		expect(testRender).toMatchSnapshot();
	});
	describe("displays", () => {
		test("title", () => {
			expect(testRender.getByText(/PROJECTS/));
		});
		test.skip("projects", () => {
			expect(testRender.getByText(""));
		});
	});
	test("contains information from json", () => {
		projectData.forEach((project) => {
			expect(testRender.getByText(project.name));
		});
	});
});