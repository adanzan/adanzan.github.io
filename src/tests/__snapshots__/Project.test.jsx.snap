// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project component matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="projects"
      >
        <h2>
          PROJECTS
        </h2>
        <div
          class="projectCards"
        >
          <div
            class="ProjectCard"
          >
            <a
              href="https://github.com/adanzan/workout-planner"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
              >
                <div
                  class="card"
                >
                  <h3>
                    Workout Planner
                  </h3>
                  <p
                    class="description"
                  >
                    A workout planner with a GUI.
                  </p>
                  <p>
                    C++, QT
                  </p>
                </div>
              </div>
            </a>
          </div>
          <div
            class="ProjectCard"
          >
            <a
              href="https://github.com/adanzan/proficiency-tracker"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
              >
                <div
                  class="card"
                >
                  <h3>
                    Proficiency Tracker
                  </h3>
                  <p
                    class="description"
                  >
                    A proficiency tracker for use in a classroom.
                  </p>
                  <p>
                    React.js, Next.js, Firebase
                  </p>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="projects"
    >
      <h2>
        PROJECTS
      </h2>
      <div
        class="projectCards"
      >
        <div
          class="ProjectCard"
        >
          <a
            href="https://github.com/adanzan/workout-planner"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
            >
              <div
                class="card"
              >
                <h3>
                  Workout Planner
                </h3>
                <p
                  class="description"
                >
                  A workout planner with a GUI.
                </p>
                <p>
                  C++, QT
                </p>
              </div>
            </div>
          </a>
        </div>
        <div
          class="ProjectCard"
        >
          <a
            href="https://github.com/adanzan/proficiency-tracker"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
            >
              <div
                class="card"
              >
                <h3>
                  Proficiency Tracker
                </h3>
                <p
                  class="description"
                >
                  A proficiency tracker for use in a classroom.
                </p>
                <p>
                  React.js, Next.js, Firebase
                </p>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
